go/vt/discovery/keyspace_events.go

```diff
@@ -329,6 +377,96 @@ func (kss *keyspaceState) onHealthCheck(th *TabletHealth) {
 	kss.ensureConsistentLocked()
 }
 
+type MoveTablesType int
+
+const (
+	MoveTablesNone MoveTablesType = iota
+	MoveTablesRegular
+	MoveTablesShardByShard
+)
+
+type MoveTablesState struct {
+	Typ   MoveTablesType
+	State MoveTablesStatus
+}
+
+func (kss *keyspaceState) getMoveTablesStatus(vs *vschemapb.SrvVSchema) (*MoveTablesState, error) {
+	mtState := &MoveTablesState{
+		Typ:   MoveTablesNone,
+		State: MoveTablesUnknown,
+	}
+
+	// if there are no routing rules defined, then movetables is not in progress, exit early
+	if (vs.RoutingRules != nil && len(vs.RoutingRules.Rules) == 0) &&
+		(vs.ShardRoutingRules != nil && len(vs.ShardRoutingRules.Rules) == 0) {
+		return mtState, nil
+	}
+
+	ctx := kss.kew.ctx
+	ts, _ := kss.kew.ts.GetTopoServer()
+
+	// collect all current shard information from the topo
+	var shardInfos []*topo.ShardInfo
+	for _, sstate := range kss.shards {
+		si, err := ts.GetShard(ctx, kss.keyspace, sstate.target.Shard)
+		if err != nil {
+			return nil, err
+		}
+		shardInfos = append(shardInfos, si)
```

Comment:
topo calls should be with a timeout, using`remote_operation_timeout` value.

Reply:
> topo calls should be with a timeout, using`remote_operation_timeout` value.
topo calls have been removed: we use the routing rules from the vschema instead of getting it again from the topo.